vcl_templates/_multivariate_tests.vcl.erb

```diff
@@ -4,58 +4,112 @@
-        set var.denominator_<%= test %>_<%= variant %> = var.denominator_<%= test %>;
+      set var.denominator_<%= test %>_<%= variant %> = var.denominator_<%= test %>;
 <% elsif variant != variants.last -%>
 <% previous_variant = variants[idx - 1] -%>
-        set var.denominator_<%= test %>_<%= variant %> = var.denominator_<%= test %>_<%= previous_variant %>;
-        set var.denominator_<%= test %>_<%= variant %> -= var.nominator_<%= test %>_<%= previous_variant %>;
+      set var.denominator_<%= test %>_<%= variant %> = var.denominator_<%= test %>_<%= previous_variant %>;
+      set var.denominator_<%= test %>_<%= variant %> -= var.nominator_<%= test %>_<%= previous_variant %>;
 <% end # if -%>
 <% end # variants.each_with_index -%>
 <% variants.each do |variant| -%>
 <% if variant == variants.first -%>
-        if (randombool(var.nominator_<%= test %>_<%= variant %>, var.denominator_<%= test %>_<%= variant %>)) {
-          set req.http.GOVUK-ABTest-<%= test %> = "<%= variant %>";
+      if (randombool(var.nominator_<%= test %>_<%= variant %>, var.denominator_<%= test %>_<%= variant %>)) {
+        set req.http.GOVUK-ABTest-<%= test %> = "<%= variant %>";
 <% elsif variant != variants.first && variant != variants.last -%>
-        } else if (randombool(var.nominator_<%= test %>_<%= variant %>, var.denominator_<%= test %>_<%= variant %>)) {
-          set req.http.GOVUK-ABTest-<%= test %> = "<%= variant %>";
+      } else if (randombool(var.nominator_<%= test %>_<%= variant %>, var.denominator_<%= test %>_<%= variant %>)) {
+        set req.http.GOVUK-ABTest-<%= test %> = "<%= variant %>";
 <% else -%>
-        } else {
-          set req.http.GOVUK-ABTest-<%= test %> = "<%= variant %>";
-        }
+      } else {
+        set req.http.GOVUK-ABTest-<%= test %> = "<%= variant %>";
+      }
 <% end -%>
 <% end # variants.each-%>
-      }
     }
+<% end # if CookielessAATest -%>
+  }
 <% end # test_config.each -%>
 <% end # ab_tests.each -%>
+}
+else {
+<% ab_tests.each do |test_config| -%>
+<% test_config.each do |test, variants| -%>
+<% if test == "CookielessAATest" -%>
+  if (table.lookup(active_ab_tests, "CookielessAATest") == "true") {
+    if (req.http.User-Agent ~ "^GOV\.UK Crawler Worker") {
+      set req.http.GOVUK-ABTest-<%= test %> = "<%= variants.last %>";
+<% variants.each do |variant| -%>
+    } else if (req.url ~ "[\?\&]ABTest-<%= test %>=<%= variant %>(&|$)") {
+      # Some users, such as remote testers, will be given a URL with a query string
+      # to place them into a specific bucket.
+      set req.http.GOVUK-ABTest-<%= test %> = "<%= variant %>";
+<% end -%>
+    } else {
+      declare local var.denominator_<%= test %> INTEGER;
+      set req.http.Client-Bucket-Full = digest.hash_md5(client.ip req.http.User-Agent);
+      set req.http.Client-Bucket-Trimmed = regsub(req.http.Client-Bucket-Full, "^([a-fA-F0-9]{10}).*$","\1");
+      set req.http.Client-Bucket = std.strtol(req.http.X-ClientIDHash-trimmed, 16);
```

Comment:
I think needs to be `std.strtol(req.http.Client-Bucket-Trimmed, 16);` given you renamed the variable above.

Reply:
👍